### PR TITLE
Fix Explorer unexpected action keyword for folder index search

### LINKED ISSUE #539: Change Actionkeyword for folder search
We should change the actionkeyword for folder search when the index search keyword is different from the path search keyword, or else the results will be weird.

### PR DESCRIPTION
**Pull request by taooceros**
fix #539

### CODE REVIEW

File: Plugins/Flow.Launcher.Plugin.Explorer/Search/ResultManager.cs [hunk 1]
```diff
@@ -39,25 +41,36 @@ internal static Result CreateFolderResult(string title, string subtitle, string
                             return false;
                         }
                     }
+                    // one of it is enabled
+                    var keyword = Settings.EnableSearchActionKeyword ? Settings.SearchActionKeyword : Settings.PathSearchActionKeyword;
+
+                    keyword = keyword == "*" ? "" : $"{keyword} ";
```
Comment (jjw24) [inline Plugins/Flow.Launcher.Plugin.Explorer/Search/ResultManager.cs:47]: Can you use Query.GlobalWildcardKeyword (whatever it's called lol) here please

File: Plugins/Flow.Launcher.Plugin.Explorer/Search/ResultManager.cs [hunk 2]
```diff
@@ -26,7 +28,7 @@ internal static Result CreateFolderResult(string title, string subtitle, string
                 TitleHighlightData = StringMatcher.FuzzySearch(query.Search, title).MatchData,
                 Action = c =>
                 {
-                    if (c.SpecialKeyState.CtrlPressed)
+                    if (c.SpecialKeyState.CtrlPressed || !(Settings.EnabledPathSearchKeyword || Settings.EnableSearchActionKeyword))
```
Comment (jjw24) [inline Plugins/Flow.Launcher.Plugin.Explorer/Search/ResultManager.cs:31]: Can you do me a favour and change Settings.EnableSearchActionKeyword all to Settings.EnabledSearchActionKeyword please

**Review by jjw24 — CHANGES_REQUESTED** [review]
good fix, see comments please

File: Plugins/Flow.Launcher.Plugin.Explorer/Search/ResultManager.cs (hunk 2 above)
Comment (jjw24) [inline Plugins/Flow.Launcher.Plugin.Explorer/Search/ResultManager.cs:31]: Is this saying both Settings.EnabledPathSearchKeyword and Settings.EnableSearchActionKeyword evaluate to false? Can we not just do (!Settings.EnabledPathSearchKeyword && !Settings.EnableSearchActionKeyword) ?

Comment (taooceros) [reply Plugins/Flow.Launcher.Plugin.Explorer/Search/ResultManager.cs:31]: If we change it to Enabled, what about move Enabled to the end of the name, like PathSearchKeywordEnabled?

Comment (taooceros) [reply Plugins/Flow.Launcher.Plugin.Explorer/Search/ResultManager.cs]: > Is this saying both Settings.EnabledPathSearchKeyword and Settings.EnableSearchActionKeyword evaluate to false? Can we not just do (!Settings.EnabledPathSearchKeyword && !Settings.EnabledSearchActionKeyword) ?

That will have two !🤣, but it seems more clear. I will change it.

Comment (jjw24) [reply Plugins/Flow.Launcher.Plugin.Explorer/Search/ResultManager.cs]: > If we change it to Enabled, what about move Enabled to the end of the name, like PathSearchKeywordEnabled?

Sorry for the late reply, I am ok moving Enabled to the end if you want to

File: Plugins/Flow.Launcher.Plugin.Explorer/Search/ResultManager.cs [hunk 3]
```diff
@@ -39,25 +41,36 @@ internal static Result CreateFolderResult(string title, string subtitle, string
...
+                    // one of it is enabled
+                    var keyword = Settings.EnabledSearchActionKeyword ? Settings.SearchActionKeyword : Settings.PathSearchActionKeyword;
+
+                    keyword = keyword == Query.GlobalPluginWildcardSign ? string.Empty : keyword + "";
 
                     string changeTo = path.EndsWith(Constants.DirectorySeperator) ? path : path + Constants.DirectorySeperator;
-                    Context.API.ChangeQuery(string.IsNullOrEmpty(query.ActionKeyword) ?
-                        changeTo :
-                        query.ActionKeyword + " " + changeTo);
+                    Context.API.ChangeQuery($"{keyword}{changeTo}");
```
Comment (jjw24) [inline Plugins/Flow.Launcher.Plugin.Explorer/Search/ResultManager.cs:50]: You removed the space here, is it meant to be added to line 47?

Comment (taooceros) [reply Plugins/Flow.Launcher.Plugin.Explorer/Search/ResultManager.cs:50]: Yes, it is supposed to be a space instead of empty string. Will fix it soom.

**Comment by jjw24** [comment]
Sorry, one more change, could we fix the typo for these
GetActionKeywordEnable
SetActionKeywordEnable

To:
GetActionKeywordEnabled
SetActionKeywordEnabled

**Comment by jjw24** [comment]
> Sorry, one more change, could we fix the typo for these
> GetActionKeywordEnable
> SetActionKeywordEnable
> 
> To:
> GetActionKeywordEnabled
> SetActionKeywordEnabled

i am got this, i am in front of comp now

**Comment by taooceros** [comment]
Sorry I was doing something else in the past hour🤣.

**Comment by jjw24** [comment]
nah all good, no probs

**Review by jjw24 — APPROVED** [review]

**Pull request merged**